### ISSUE TITLE
accessibilityRole="heading" seems to cause problems on Android

### ISSUE THREAD

**Issue by BogdanCerovac** [expo/expo#20713]
### Summary

When I add the `accessibilityRole="heading"` to `<Text>` the whole app won't render and is unable to refresh/update. Same issue when using real or virtual device with Android. Changing SDKs didn't help, updated Expo app on phones, didn't help...

iOS worked well. Web as well.

Seems that problem is directly connected to adding `accessibilityRole="heading"`. Console clean, no issues seen.

I made a [video that demonstrates the problem](https://youtu.be/MVd4WHjjkHA).

### What platform(s) does this occur on?

Android

### Environment

Used Snack online

### Minimal reproducible example

Link to Snack: https://snack.expo.dev/agJkMBIPY

Code for reference;

```export default function App() {
  return (
    <View style={styles.container}>
      <Text accessibilityRole="heading" style={styles.heading}>
        Application 
      </Text>
     <Text style={styles.paragraph}>
        Lorem ipsum and so on...
      </Text>
    </View>
  );
}```

**Comment by BogdanCerovac** [comment]
The issue is that I used "heading" instead of "header" due to old reference.

Using `accessibilityRole="header"` works fine.

Would be nice to inform developers using obsolete / not available accessibilityRole values in some way...
